### PR TITLE
Fix #1802 : Reading text size not scrollable or flexible

### LINKED ISSUE #1802: Reading text size not scrollable or flexible
**Describe the bug**
Reading Text Size screen not fitting the flexible text

**To Reproduce**
1. Open reading text size screen from options
2. Change text size and see the text size changing above

**Expected behavior**
The complete text should be visible

Mock link - https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/2eaf085a-c2a7-4b1d-b515-5b32f8642669/specs/

**Screenshots**
Screenshot_2020-09-04-19-03-03-426_org oppia app

**Device**
- Device/emulator being used
- SDK version

**Additional context**

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes #1802 
Preview Text now displays the correct and real size of the text in stories.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Screenshots for the same -

### CODE REVIEW

**Comment by prayutsu** [comment]
@anandwana001 PTAL

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt [hunk 1]
```diff
@@ -44,23 +45,31 @@ class ReadingTextSizeFragmentPresenter @Inject constructor(private val fragment:
...
         binding.readingTextSizeSeekBar.progress = 5
-        binding.previewTextview.textSize =
-          getReadingTextSizeInFloat(ReadingTextSize.MEDIUM_TEXT_SIZE)
+        binding.previewTextview.setTextSize(
+          TypedValue.COMPLEX_UNIT_PX,
+          getReadingTextSizeInFloat(ReadingTextSize.SMALL_TEXT_SIZE)
+        )
       }
-      getReadingTextSize(ReadingTextSize.LARGE_TEXT_SIZE) -> {
+      getReadingTextSize(ReadingTextSize.MEDIUM_TEXT_SIZE) -> {
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt]: this is mistaken here, I guess. It should not be changed

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt [hunk 2]
```diff
@@ -44,23 +45,31 @@ class ReadingTextSizeFragmentPresenter @Inject constructor(private val fragment:
     when (readingTextSize) {
       getReadingTextSize(ReadingTextSize.SMALL_TEXT_SIZE) -> {
         binding.readingTextSizeSeekBar.progress = 0
-        binding.previewTextview.textSize =
+        binding.previewTextview.setTextSize(
+          TypedValue.COMPLEX_UNIT_PX,
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt:49]: we are using the normal sp values than why we need to mention pixel value here?
Could you explain this and also add why we can't fix this from UI side?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Check few comments @prayutsu 

At the very first time when you open the reading text size screen and you change from medium to small the text size doesn't change. Also, isn't this the text size looking too small?

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt (hunk 2 above)
Comment (prayutsu) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt:49]: Please have a look [here](https://stackoverflow.com/questions/20364993/setting-textsize-programmatically/20365110).

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt (hunk 1 above)
Comment (prayutsu) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt]: yes, this is mistaken, I'll correct it.

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt (hunk 2 above)
Comment (rt4914) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt:49]: Screenshot 2020-09-08 at 10 44 37 AM

Based on the documentation of this function, it does make sense.

Comment (prayutsu) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt:49]: Actually, I logged out the value of defaultReadingTextSizeInFloat (in getReadingTextSIzeInFloat function), it is set to 18sp in dimens.xml but when I logged its value, it was showing 51.875, so, it was inferred in pixels, so when I am setting textSize, I am actually explicitly mentioning that the textSize passed as the parameter is actually in pixels.

**Comment by prayutsu** [comment]
> Check few comments @prayutsu
> 
> At the very first time when you open the reading text size screen and you change from medium to small the text size doesn't change. Also, isn't this the text size looking too small?

I think this will get fixed when I correct that mistaken "MEDIUM_TEXT_SIZE" mentioned by you.

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt (hunk 2 above)
Comment (anandwana001) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt:49]: Thanks for looking into deep. This is good to go now.

File: app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt (hunk 1 above)
Comment (anandwana001) [reply app/src/main/java/org/oppia/app/options/ReadingTextSizeFragmentPresenter.kt]: we are good to merge after this update

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @prayutsu

**Pull request merged**